### PR TITLE
Node-ify lifetime

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

File: rust.ungram [hunk 1]
```diff
@@ -456,13 +459,13 @@ WhileExpr =
   loop_body:BlockExpr
 
 Label =
-  'lifetime'
+  'lifetime_ident' ':'
```
Comment (matklad) [inline rust.ungram:462]: This probably wants to be `Lifetime` as well.

**Comment by matklad** [comment]
Could check (using `[patch.'crates-io']`) that this works with rust-analyzer? MIght make sense to prep a draft PR

**Comment by Veykril** [comment]
I'm almost certain this will cause problems in RA, would surprise me if not 😄 Sure I can start working on a PR that pulls this in properly first before this gets merged.

**Comment by matklad** [comment]
bord d+

**Comment by Veykril** [comment]
you typo'd bors

**Comment by matklad** [comment]
bors d+

**Pull request merged**